src/app_unit.rs

```diff
@@ -289,6 +289,92 @@ impl Au {
     }
 }
 
+#[cfg(feature = "num_traits")]
+impl num_traits::ToPrimitive for Au {
+    // Float conversions
+
+    #[inline]
+    fn to_f32(&self) -> Option<f32> {
+        Some(self.to_f32_px())
+    }
+
+    #[inline]
+    fn to_f64(&self) -> Option<f64> {
+        Some(self.to_f64_px())
+    }
+
+    // Signed int conversions
+
+    #[inline]
+    fn to_i8(&self) -> Option<i8> {
+        None
+    }
+
+    #[inline]
+    fn to_i16(&self) -> Option<i16> {
+        None
+    }
+
+    #[inline]
+    fn to_i32(&self) -> Option<i32> {
+        Some(self.to_px())
+    }
+
+    #[inline]
+    fn to_i64(&self) -> Option<i64> {
+        Some(self.to_px() as i64)
+    }
+
+    #[inline]
+    fn to_i128(&self) -> Option<i128> {
+        Some(self.to_px() as i128)
+    }
+
+    #[inline]
+    fn to_isize(&self) -> Option<isize> {
+        Some(self.to_px() as isize)
+    }
+
+    // Unsigned int conversions
+
+    #[inline]
+    fn to_u8(&self) -> Option<u8> {
+        None
+    }
+
+    #[inline]
+    fn to_u16(&self) -> Option<u16> {
+        None
+    }
+
+    #[inline]
+    fn to_u32(&self) -> Option<u32> {
+        None
+    }
+
+    #[inline]
+    fn to_u64(&self) -> Option<u64> {
+        None
+    }
+
+    #[inline]
+    fn to_u128(&self) -> Option<u128> {
+        None
+    }
+
+    #[inline]
+    fn to_usize(&self) -> Option<usize> {
+        None
+    }
```

Comment:
It looks like you could simply implement `to_u64` and then the other implementations have defaults?

Reply:
Yes, I could do. Did it this way to be explicit. But I can go down that route if you prefer?